client/lib/i18n-utils/utils.js

```diff
@@ -110,18 +110,130 @@ export function getLocaleFromPath( path ) {
  * @returns {string} original path with new locale slug
  */
 export function addLocaleToPath( path, locale ) {
-	const urlParts = parse( path );
+	const urlParts = url.parse( path );
 	const queryString = urlParts.search || '';
 
 	return removeLocaleFromPath( urlParts.pathname ) + `/${ locale }` + queryString;
 }
 
-export function addLocaleToWpcomUrl( url, locale ) {
-	if ( locale && locale !== 'en' ) {
-		return url.replace( 'https://wordpress.com', 'https://' + locale + '.wordpress.com' );
+const localesToSubdomains = {
+	'pt-br': 'br',
+	br: 'bre',
+	zh: 'zh-cn',
+	'zh-hk': 'zh-tw',
+	'zh-sg': 'zh-cn',
+	kr: 'ko',
+};
+
+const magnificentNonEnLocales = [
```

Comment:
Does it make sense to use the [config value](https://github.com/Automattic/wp-calypso/blob/master/config/_shared.json#L44), `magnificent_non_en_locales`?

Reply:
It does indeed, thanks for the pointer! I have updated the code.